Feat: initial implementation as config (#2)
moxystudio/jest-config: README.md

## Installation

```sh
$ npm install --save-dev @moxy/jest-config
```

## Usage

Create `jest.config.js` at the root of your project:

```js
const { baseConfig } = require('@moxy/jest-config')

module.exports = baseConfig();
```

## Addons

This packages comes with extra addons that further tweak the base Jest configuration to cover the needs of common situations. Here's a list of all addons we offer so far:

| Addon | Description |
| --- | --- |
| [withWeb](addons/withWeb/withWeb.js) | Adds setup and ignore patterns we use in [`next-with-moxy`](https://www.github.com/moxystudio/next-with-moxy) |

To use addons, use the `compose` function that comes with this package. **Keep in mind**, the first item should always be the default configuration, `baseConfig`! Here's an example of using `compose`:

```js
const { compose, baseConfig, withWeb } = require('@moxy/jest-config');

module.exports = compose([baseConfig, withWeb]);
```

If you want to make your own addon, you can! It just needs to have the following structure:

```js
const { compose, baseConfig } = require('@moxy/jest-config');

// Receives previous configuration
const withAddon = (configuration) => {

> 

Reviewer: Can you remove this empty line?

    // Add your options to configuration
    // For example, do not test '.data.js' files

>     configuration.testPathIgnorePatterns = ['/.*.data.js$/']

Reviewer: Missing ;

    // Returns configuration with added options
    return configuration;

> }

Reviewer: Missing ;

# jest-config

[![NPM version][npm-image]][npm-url] [![Downloads][downloads-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Coverage Status][codecov-image]][codecov-url] [![Dependency status][david-dm-image]][david-dm-url] [![Dev Dependency status][david-dm-dev-image]][david-dm-dev-url]

[npm-url]:https://npmjs.org/package/@moxy/jest-config
[downloads-image]:https://img.shields.io/npm/dm/@moxy/jest-config.svg
[npm-image]:https://img.shields.io/npm/v/@moxy/jest-config.svg
[travis-url]:https://travis-ci.org/moxystudio/jest-config
[travis-image]:http://img.shields.io/travis/moxystudio/jest-config/master.svg
[codecov-url]:https://codecov.io/gh/moxystudio/jest-config
[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/jest-config/master.svg
[david-dm-url]:https://david-dm.org/moxystudio/jest-config
[david-dm-image]:https://img.shields.io/david/moxystudio/jest-config.svg
[david-dm-dev-url]:https://david-dm.org/moxystudio/jest-config?type=dev
[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/jest-config.svg

MOXY's Jest configuration to be used across several JavaScript projects.

## Installation

```sh

> $ npm install @moxy/jest-config

Reviewer: `--save-dev`

```

## Usage

Create `jest.config.js` at the root of your project:

```js

> const { jestConfig } = require('@moxy/jest-config')

Reviewer: Perhaps this should be named `baseConfig` or `createBaseConfig`?

module.exports = jestConfig();
```

## Addons

> This package comes with some preset addons that add further options to the Jest configuration to cover the needs of common situations. Here's a list of all addons we offer so far:

Reviewer: > This packages comes with extra addons that further tweak the base Jest configuration to ....

| Addon | Description |

> |  ---   |     ---     |

Reviewer: Let's align the table.

| [withWeb](addons/withWeb/withWeb.js)   | Adds setup and ignore patterns we use in [`next-with-moxy`](https://www.github.com/moxystudio/next-with-moxy) |

> ### Instructions

Reviewer: Perhaps this header is not necessary at all

> 

Reviewer: Remove extra line

> To use addons, use the `compose` function offered with this package. **Keep in mind**, the first item should always be the default configuration, `jestConfig`! Here's an example of using `compose`:

Reviewer: ```suggestion
To use addons, use the `compose` function that comes with this package. **Keep in mind**, the first item should always be the base configuration! Here's an example that uses `compose`:
```

```js
const { compose, jestConfig, withWeb } = require('@moxy/jest-config');

module.exports = compose([jestConfig, withWeb]);
```

If you want to make your own addon, you can! It just needs to have the following structure:

```js
const { compose, jestConfig } = require('@moxy/jest-config');

// Receives previous configuration

> function myAddon(configuration) {

Reviewer: Change to arrow function and keep the `withXXX` naming to be inline with the rest.

    // Add your options to configuration
    // For example, do not test '.data.js' files
    configuration.testPathIgnorePatterns = ['/.*.data.js$/']

    // Returns configuration with added options
    return configuration;
}

module.exports = compose([jestConfig, myAddon]);
```

### Without addons

> If you want to add your own options without using our addons or making your own, you must import this configuration and change the returned object, which you must then export.

Reviewer: ```suggestion
If you want to add your own options without using our addons or making your own, you may imperatively change the returned base config like so:
```

```

## Addons

This packages comes with extra addons that further tweak the base Jest configuration to cover the needs of common situations. Here's a list of all addons we offer so far:

| Addon | Description |
| --- | --- |
| [withWeb](addons/withWeb/withWeb.js) | Adds setup and ignore patterns we use in [`next-with-moxy`](https://www.github.com/moxystudio/next-with-moxy) |

To use addons, use the `compose` function that comes with this package. **Keep in mind**, the first item should always be the default configuration, `baseConfig`! Here's an example of using `compose`:

```js
const { compose, baseConfig, withWeb } = require('@moxy/jest-config');

module.exports = compose([baseConfig, withWeb]);
```

If you want to make your own addon, you can! It just needs to have the following structure:

```js
const { compose, baseConfig } = require('@moxy/jest-config');

// Receives previous configuration
const withAddon = (configuration) => {
    // Add your options to configuration
    // For example, do not test '.data.js' files
    configuration.testPathIgnorePatterns = ['/.*.data.js$/'];

    // Returns configuration with added options
    return configuration;

> }

Reviewer: ```suggestion
};
```
